src/org/zaproxy/zap/extension/ascanrulesBeta/HPP.java

```diff
@@ -154,7 +157,12 @@ public void scan() {
 					log.debug("Injecting payload...");
 					HttpMessage newMsg = getNewMsg();
 					newMsg.setGetParams(tags);
-					sendAndReceive(newMsg);
+					try {
+						sendAndReceive(newMsg);
+					} catch (IllegalStateException|UnknownHostException ex) {
+						log.info("Caught " + ex.getClass().getName() + " when accessing: " + newMsg.getRequestHeader().getURI().toString() + 
+								"\n The target may have replied with a poorly formed redirect due to our input.");
```

Comment:
Right, was saying that the response used in following checks would not correspond to the attack request (not that the response that lead to the exception was not a result of the attack).
btw, it might worth also logging the exception message as that contains more details of the problem.